src/actions.js

```diff
@@ -75,76 +74,71 @@ export function addCheckResult(
 // ASYNC (THUNK) ACTIONS.
 
 // Fetching the statuses.
-export function requestStatus(version: ?string): ThunkAction<void> {
+export function requestStatus(version: ?string) {
   const notifyChanges = (checkTitle, status) => {
     if (Notification.permission === 'granted') {
       new Notification(`${checkTitle} : status changed (${status}).`);
     }
   };
 
-  return function(dispatch: Dispatch, getState: GetState) {
+  return async function(dispatch: Dispatch, getState: GetState) {
     const versionToCheck = version || getState().version;
     if (!versionToCheck) {
       return;
     }
     // Save previous results so we can check if something changed.
     const prevResults = getState().checkResults;
     dispatch(setVersion(versionToCheck));
-    getReleaseInfo(versionToCheck)
-      .then(releaseInfo => {
-        dispatch(updateReleaseInfo(releaseInfo));
-        releaseInfo.checks.map((check: CheckInfo) => {
-          return checkStatus(check.url).then(result => {
-            // Detect if the result changed, and notify!
-            const prevResult = prevResults[check.title];
-            if (prevResult && prevResult.status !== result.status) {
-              notifyChanges(check.title, result.status);
-            }
-            dispatch(addCheckResult(check.title, result));
-          });
-        });
-      })
-      .catch((err: string) => {
-        console.error(
-          `Failed getting the release info for ${versionToCheck}`,
-          err,
-        );
+    const releaseInfo = await getReleaseInfo(versionToCheck);
+    dispatch(updateReleaseInfo(releaseInfo));
+    const checks = releaseInfo.checks.map((check: CheckInfo) => {
+      return checkStatus(check.url).then(result => {
+        const prevResult = prevResults[check.title];
```

Comment:
```js
const result = await checkStatus(check.url); 
```

or am I missing something ?

Reply:
Maybe I misunderstood your question, but have you read the thread here? https://github.com/mozilla/delivery-dashboard/pull/63#pullrequestreview-61800246

It seems that what you're proposing is the solution I had implemented at first, but then I rolled it back to use `Promise.all` so we won't risk having errors being silenced/dropped. Or are you saying that they won't be silenced/dropped?

Reply:
They're only silenced/dropped if you don't await them. Rolling back to your previous version, but grouping all of the promises in this `checks` variable, and then awaiting them all using `Promise.all`, should probably check all the boxes.

Reply:
Ok I gave it a try @glasserc please let me know if that's what you were thinking about ;)